pkg/kubectl/cmd/portforward.go

```diff
@@ -43,31 +44,58 @@ $ kubectl port-forward -p mypod 0:5000`
 )
 
 func NewCmdPortForward(f *cmdutil.Factory) *cobra.Command {
+	pfw := portForwardCmd{
+		factory: &defaultForwarderFactory{},
+	}
 	cmd := &cobra.Command{
 		Use:     "port-forward -p POD [LOCAL_PORT:]REMOTE_PORT [...[LOCAL_PORT_N:]REMOTE_PORT_N]",
 		Short:   "Forward one or more local ports to a pod.",
 		Long:    "Forward one or more local ports to a pod.",
 		Example: portforward_example,
 		Run: func(cmd *cobra.Command, args []string) {
-			err := RunPortForward(f, cmd, args)
+			err := pfw.validateParams(cmd, args)
+			cmdutil.CheckErr(err)
+			err = pfw.runPortForward(f, args)
 			cmdutil.CheckErr(err)
 		},
 	}
-	cmd.Flags().StringP("pod", "p", "", "Pod name")
+	cmd.Flags().StringVarP(&pfw.pod, "pod", "p", "", "Pod name")
 	cmd.MarkFlagRequired("pod")
 	// TODO support UID
 	return cmd
 }
 
-func RunPortForward(f *cmdutil.Factory, cmd *cobra.Command, args []string) error {
-	podName := cmdutil.GetFlagString(cmd, "pod")
-	if len(podName) == 0 {
+type portForwardCmd struct {
```

Comment:
Only reason is to test... abstract out call to portforward.New(...) and following call to that object's ForwardPorts method. I'll think of a way to simplify it.